daffodil-runtime2/src/test/scala/org/apache/daffodil/runtime2/TestCodeGenerator.scala

```diff
@@ -78,35 +91,29 @@ class TestCodeGenerator {
     val cg = pf.forLanguage("c")
 
     // Generate code from the test schema successfully
-    val outputDir = cg.generateCode(None, "./generateCode_tmp")
+    val outputDir = cg.generateCode(None, s"$tmpOutputDir")
```

Comment:
Thoughts on having each test create it's own temp directory? That way there's no possibility for one test to interfere with another test?

I've also wished for a while that we had a function that all our tests could use so we could have a standard way to create temp directories. Right now lots of tests do it differently and we pollute /tmp with a bunch of files/directories. It would be nice if had something like this instead:

```scala
object DaffodilUtil {
  private val rootTempDir = Paths.get(System.getProperty("java.io.tmpdir"), "daffodil")
  def withDaffodilTempDir[T](prefix: String = "")(body: Path => T): T = {
      Files.createDirectories(rootTempDir)
      val tmpDir = Files.createTempDirectory(rootTempDir, prefix)
      try {
        body(tmpDir)
      } finally {
        // recurisively delete tmpDir
      }
  }
}
```

Then, we can update individual tests, and maybe even things like the TDML runner, that need a temp directory can just do something like:
```scala
DaffodilUtil.withTempDir() { tmpDir => 
  // do test using tmpDir, it will automatically be deleted when complete
}
```

Might be out of scope for this change though. If you don't want to add somethign like this to this PR, I'll create a bug so we don't forget. I've had this in the back of my head for a while.

Reply:
Nice idea, although I've simply added calls to os.temp.dir() so each test method creates its own temp directory and kept using `@After` to remove the temp directory.  (The os-lib documentation says the directory created by os.temp.dir() is supposed to be removed when the JVM exits, but I've never found that to happen in practice, bummer.)